enhancements/two-nodes-openshift/2no.md

```diff
@@ -0,0 +1,449 @@
+
+1. Risk: Rebooting the surviving peer would require human intervention before the cluster starts, increasing downtime and creating an admin burden at remote sites
+   1. Mitigation: Lifecycle events, such as upgrades and applying new `MachineConfig`s, are not permitted in a single-node degraded state
+   1. Mitigation: Usage of the MCO Admin Defined Node Disruption [feature](https://github.com/openshift/enhancements/pull/1525) will further reduce the need for reboots.
+   1. Mitigation: The node will be reachable via SSH and the confirmation can be scripted
+   1. Mitigation: It may be possible to identify scenarios where, for a known hardware topology, it is safe to allow the node to proceed automatically.
+
+1. Risk: “Something changed, let's reboot” is somewhat baked into OCP’s DNA and has the potential to be problematic when nodes are actively watching for their peer to disappear, and have an obligation to promptly act on that disappearance by power cycling them.
+   1. Mitigation: Identify causes of reboots, and either avoid them or ensure they are not treated as failures.
+   This may require an additional enhancement.
+
+1. Risk: We may not succeed in identifying all the reasons a node will reboot
+   1. Mitigation: ... testing? ...
+
+1. Risk: This new platform will have a unique installation flow
+   1. Mitigation: ... CI ...
+
+
+
+
+### Drawbacks
+
+The two-node architecture represents yet another distinct install type for users to choose from.
+
+The existence of 1, 2, and 3+ node control-plane sizes will likely generate customer demand to move between them as their needs change.
+Satisfying this demand would come with significant technical and support overhead.
+
+## Open Questions [optional]
+
+1. Are there any normal lifecycle events that would be interpreted by a peer as a failure, and where the resulting "recovery" would create unnecessary downtime?
+   How can these be avoided?
+2. Is there are requirement for disconnected cluster support in the initial release?
+3. Are there consequences of changing the parentage of processes running cri-o, kubelet, and etcd? (E.g. user process limits)
+4. In the test plan, which subset of layered products needs to be evaluated for the initial release (if any)?
+5. How are the BMC credentials getting from the install-config and onto the nodes?
+6. How does the cluster know it has achieved a ready state?
+   From the cluster's perspective, we know that CEO needs to have `managedEtcd: External` set, and all of the operators need to be available. However, there is a time delay between when that configuration is set by CEO and when etcd comes back up as health after it's restarted under the RHEL-HA components. Is a fixed wait enough to determine that we have successfully transitioned to the new topology? If not, how do we detect this?
+7. Are there any scenarios that would require signaling from the cluster to the RHEL-HA components to modify or change their behavior?
```

Comment:
Depends if, in future versions, we want any part of that configuration to be accessible to end-users.
Possibly also depending on how we handle graceful reboots.